src/main/java/org/candlepin/paging/PagingUtil.java

```diff
@@ -153,4 +189,82 @@ public Stream<T> applyPaging(Collection<T> collection) {
             this.applyPaging(Stream.of(), 0);
     }
 
+
+    // TODO: This could use some nicer, cleaner integration; it's rather bolted on at the moment.
+
+    /**
+     * Applies any paging within the request context to the specified query builder, overwriting any paging
+     * or sorting values that may have already been applied. If no paging has been defined in the request
+     * context, this method will not make any changes to the provided query builder; but it will still
+     * validate the resultant element count against the maximum configured page size. If the given query
+     * builder is null, this method returns null.
+     *
+     * @param queryBuilder
+     *  the query builder to which any request context paging should be applied
+     *
+     * @throws BadRequestException
+     *  if the number of results the query builder will return exceeds the maximum page size
+     *
+     * @return
+     *  the provided query builder
```

Comment:
We may want to update this return statement and the the rest of the javadoc to mention that we are executing the query and returning a stream of entities.

Reply:
Whoops. You can see echoes of the original approach there before I changed my mind.